internal/controller/controllers/crd_utils.go

```diff
@@ -0,0 +1,69 @@
+package controllers
+
+import (
+	"context"
+
+	adiiov1alpha1 "github.com/openshift/assisted-service/internal/controller/api/v1alpha1"
+	"github.com/sirupsen/logrus"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type CRDUtils struct {
+	client client.Client
+}
+
+func NewCRDUtils(client client.Client) *CRDUtils {
+	return &CRDUtils{client: client}
+}
+
+// CreateAgentCR Creates an Agent CR representing a Host/Agent.
+// If the Host is already registered, the CR creation will be skipped.
+// if the Cluster was not created via K8s API, then the Cluster NameSpace will be empty and Agent CR creation will be skipped
+func (u *CRDUtils) CreateAgentCR(ctx context.Context, log logrus.FieldLogger, hostId, clusterNamespace, clusterName string) error {
```

Comment:
Why isn't this part of agent_controller?

Reply:
The flow is that Assisted Service is creating the CR when a host register, therefore the controller is not involved.